notify/notify.go

```diff
@@ -65,6 +65,11 @@ type Integration struct {
 	rs       ResolvedSender
 	name     string
 	idx      int
+
+	mtx                *sync.RWMutex
+	lastError          error
+	lastNotify         time.Time
+	lastNotifyDuration time.Duration
```

Comment:
These 3 fields are used together everywhere. I think it makes sense to create a structure instead.

Reply:
I don't know if referencing these fields together twice (in the report related functions) deserves adding another level of abstraction.